website/static/js/formViewModel.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Maintains the base class for knockoutJS form ViewModels
+ */
+'use strict';
+
+var ko = require('knockout');
+
+var $osf = require('osfHelpers');
+var oop = require('js/oop');
+
+var ValidationError = oop.extend(Error, {
+    constructor: function (messages, header, level) {
+        this.super.constructor();
+        this.messages = messages || [];
+        this.header = header || 'Error';
+        this.level = level || 'warning';
+    }
+});
+
+var FormViewModel = oop.defclass({
+    constructor: function() {
+        var self = this;
+    },
+    isValid: function() {
+        throw new Error('FormViewModel subclass must implement isValid');
+    },
+    submit: function() {
+        try {
+            this.isValid();
+            return true; // Allow form to submit normally
+        } catch (ValidationError) {
```

Comment:
This syntax is incorrect. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch